server/src/pages/shot/text-tool.js

```diff
@@ -0,0 +1,205 @@
+const React = require("react");
+const PropTypes = require("prop-types");
+const { Localized } = require("fluent-react/compat");
+const sendEvent = require("../../browser-send-event.js");
+const { Selection } = require("../../../shared/selection");
+const { ColorPicker } = require("./color-picker");
+
+// Text input height and padding is determined by text size
+// See frame.scss mixin type-box($size)
+// TEXT_INPUT_PADDING_TOP is top padding and
+// TEXT_INPUT_PADDING_LEFT is multiplier used to calculate left padding
+const TEXT_INPUT_PADDING_TOP = 4;
+const TEXT_INPUT_PADDING_LEFT = 0.4;
+const TEXT_INPUT_HEIGHT = 1.3;
+
+// Padding used by draggable div
+const DRAG_PADDING = 10;
+const FONT_STYLE = "sans-serif";
+const FONT_WEIGHT = 900;
+const INIT_FONT_SIZE = 36;
+
+exports.TextTool = class TextTool extends React.Component {
+  constructor(props) {
+    super(props);
+    this.textInput = React.createRef();
+
+    this.canvasCssWidth = props.canvasCssWidth;
+    this.canvasCssHeight = props.canvasCssHeight;
+
+    const INIT_LEFT = (this.canvasCssWidth / 2) - INIT_FONT_SIZE * TEXT_INPUT_PADDING_LEFT;
+    const INIT_TOP = (this.canvasCssHeight / 2) - ((INIT_FONT_SIZE * TEXT_INPUT_HEIGHT) / 2);
+
+    this.state = {
+      left: INIT_LEFT,
+      top: INIT_TOP,
```

Comment:
These two values do not need to be in the component state, especially since they never change.

Reply:
Ah sorry, I see where they are updated now.  Please ignore ⬆️